Kudu.Services.Web/App_Start/NinjectServices.cs

```diff
@@ -450,7 +450,7 @@ public static void RegisterRoutes(IKernel kernel, RouteCollection routes)
             routes.MapHttpRouteDual("zip-put-files", "zip/{*path}", new { controller = "Zip", action = "PutItem" }, new { verb = new HttpMethodConstraint("PUT") });
 
             // Zip push deployment
-            routes.MapHttpRoute("zip-push-deploy", "api/zipdeploy", new { controller = "PushDeployment", action = "ZipPushDeploy" }, new { verb = new HttpMethodConstraint("POST") });
+            routes.MapHttpRouteDual("zip-push-deploy", "zipdeploy", new { controller = "PushDeployment", action = "ZipPushDeploy" }, new { verb = new HttpMethodConstraint("POST", "PUT") });
```

Comment:
Do we really want both routes to do both things? When it comes to routes, having more ways of doing the same thing is not necessarily a good thing.

Reply:
we can't break `/api/zipdeploy`, are you saying we should leave `/api/zipdeploy` to only handle binary stream and have `/zipdeploy` handle only JSON? this change makes both `/zipdeploy` and `/api/zipdeploy` handles both scenarios.

Reply:
> are you saying we should leave `/api/zipdeploy` to only handle binary stream and have `/zipdeploy` handle only JSON?

Yes, that's what I meant. But not a big thing, I'm fine either way on this one.